tmk_core/common/keyboard.c

```diff
@@ -64,20 +64,41 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 
 #ifdef MATRIX_HAS_GHOST
+static uint16_t matrix_ghost_check[MATRIX_ROWS];
 static bool has_ghost_in_row(uint8_t row)
 {
-    matrix_row_t matrix_row = matrix_get_row(row);
-    // No ghost exists when less than 2 keys are down on the row
-    if (((matrix_row - 1) & matrix_row) == 0)
+    matrix_row_t matrix_row = (matrix_get_row(row) & matrix_ghost_check[row]);
+    /* No ghost exists when less than 2 keys are down on the row.
+    If there are "active" blanks in the matrix, the key can't be pressed by the user,
+    there is no doubt as to which keys are really being pressed.
+    The ghosts will be ignored, they are KC_NO.   */
+    if (((matrix_row - 1) & matrix_row) == 0){
         return false;
-
-    // Ghost occurs when the row shares column line with other row
+    }
+    // Ghost occurs when the row shares column line with other row, blanks in the matrix don't matter
+    // If there are more than two real keys pressed and they match another row's real keys, the row will be ignored.
     for (uint8_t i=0; i < MATRIX_ROWS; i++) {
-        if (i != row && (matrix_get_row(i) & matrix_row))
+        if (i != row && __builtin_popcount((matrix_get_row(i) & matrix_ghost_check[i]) & matrix_row) > 1){
```

Comment:
Yes, sorry, my fault. That's what happens when trying to analyze a hardware algorithm from a software point of view.

It does indeed seem to work as it should and also handles different combinations of disabled keys correctly. In fact with this change the other if statement wouldn't really be needed, since your newly added if wouldn't be triggered unless there are more than one key active in a row.